modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BaseEditor.js

```diff
@@ -93,14 +103,27 @@ const BaseEditor = forwardRef(
 			};
 		}, [contents, getHTML, name]);
 
-		return (
+		useEffect(() => {
+			getEditorConfigurationCX(config).then((config) => {
+				setEditorConfiguration(config);
+				setCXLoaded(true);
+			});
+		}, [config]);
```

Comment:
I'd keep frontend CX implementation completely separate from this PR, especially if we're adding a substantial backend logic. The end point of a PR with backend handling can be CX `URL` in editor props. Then, as a separate task, we can implement resolution of CX request. I'd assume this would include new [public API entry](https://github.com/liferay/liferay-frontend-projects/tree/master/projects/js-toolkit/packages/js-api) for CKEditor or CKEditor4, where we can consider opacity of `config.`

Reply:
If I undestood, do you mean to wrap the <CKeditor> to handle the CX loading outside of it? somewhere around [here](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor_classic.jsp#L24-L36) maybe?

Reply:
I'd do something similar to what we doing in FDS, with [loadClientExtensions](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L237-L303). It can be much simpler with CKEditor, as there is only one CX.

Reply:
oh wow that code is not in the branch I'm working on. I'll update right now and see if API matches better with the starting point I took to load the CX. Nice :tada: